stingray/pulse/modeling.py

```diff
@@ -0,0 +1,180 @@
+
+    Other Parameters
+    ----------
+    amplitude : float
+        the value for x=mean
+    mean : float
+        mean of the sinc function
+    width : float
+        width of the sinc function
+
+    Returns
+    -------
+    sqvalues : array-like
+         Return square of sinc function
+    """
+    sqvalues = amplitude * sinc((x-mean)/width) ** 2
+    return sqvalues
+
+
+def sinc_square_deriv(x, amplitude=1., mean=0., width=1.):
+    """
+    Calculate partial derivatives of sinc-squared.
+
+    Parameters
+    ----------
+    x: array-like
+
+    Other Parameters
+    ----------
+    amplitude : float
+        the value for x=mean
+    mean : float
+        mean of the sinc function
+    width : float
+        width of the sinc function
+
+    Returns
+    -------
+    d_amplitude : array-like
+         partial derivative of sinc-squared function
+         with respect to the amplitude
+    d_mean : array-like
+         partial derivative of sinc-squared function
+         with respect to the mean
+    d_width : array-like
+         partial derivative of sinc-squared function
+         with respect to the width
+    """
+
+    x_is_zero = x == mean
+
+    d_x = 2 * amplitude * sinc((x-mean)/width) * (x * np.cos((x-mean)/width) \
+        - np.sin((x-mean)/width)) / ((x-mean)/width)**2
+    d_amplitude = sinc((x-mean)/width)**2
+    d_mean = d_x*(-1/width)
+    d_width = d_x*(-(x-mean)/(width)**2)
+
+    d_x[x_is_zero] = 0
+
+    return [d_amplitude, d_mean, d_width]
+
+
+SincModel = models.custom_model(sinc_square_model,
+                                fit_deriv=sinc_square_deriv)
+
+
+def fit_sinc(x, y, amp=1.5, mean=0., width=1., tied={}, fixed={}, bounds={},
+             obs_length=None):
+    """
+    Fit a sinc function to x,y values.
+
+    Parameters
+    ----------
+    x : array-like
+    y : array-like
+
+    Other Parameters
+    ----------------
+    amp : float
+        The initial value for the amplitude
+    mean : float
+        The initial value for the mean of the sinc
+    obs_length : float
+        The length of the observation. Default None. If it's defined, it
+        fixes width to 1/(pi*obs_length), as expected from epoch folding 
+        periodograms
+    width : float
+        The initial value for the width of the sinc. Only valid if 
+        obs_length is 0
+    tied : dict
+    fixed : dict
+    bounds : dict
+        Parameters to be passed to the astropy models
+        http://docs.astropy.org/en/stable/api/astropy.modeling.functional_models.Gaussian1D.html
+
+    Returns
+    -------
+    sincfit : function
+        The best-fit function, accepting x as input
+        and returning the best-fit model as output
+    """
+    if obs_length is not None:
+        width = 1 / (np.pi * obs_length)
+        fixed["width"] = True
+
+    sinc_in = SincModel(amplitude=amp, mean=mean,width=width, tied=tied,
+                        fixed=fixed, bounds=bounds)
+    fit_sinc = fitting.LevMarLSQFitter()
```

Comment:
To use it to full power, we should have an idea of the likelihood to use. However, as far as I understand this might get complicated as epoch folding has a chi squared distribution with nbin -1 dof, the z^2_n with n dof, etc. Astropy's default fitting is extremely easy to use for a newcomer, then we might think carefully how to use the `stingray.modeling` functions (in a PR by one of us)?